### ISSUE TITLE
The Players will lose their items, if placed inside Enchanting Table or Anvil secondary slot. [Failed to handle packet ServerboundContainerButtonClickPacket]

### ISSUE THREAD

**Issue by cptDeeJay** [IzzelAliz/Arclight#1518]
### I have confirmed that ...

- [X] Arclight is up to date
- [X] all dependencies are installed
- [X] all plugins and mods are up to date
- [X] unable to reproduce in Spigot
- [X] unable to reproduce in Forge

### Arclight version

arclight-neoforge-1.21.1-1.0.0-SNAPSHOT

### OS & Java versions

CasaOS (Ubuntu) java version 21.0.3

### Plugins and Mods

```raw
CrisTreeCapitator- disabled by changing extension
No other plugins or mods
```

### Description

It has been found that on lastest Arclight version **1.21.1(Feudal Kings)** with commit **`d0dda85`** The Players might or will lose their items, if placed inside Enchanting Table or Anvil secondary slot, with error message appear in terminal. Please note that no plugin or mod is activated on the server and the issue occurs for both, Players that are connected by local network and for the Players that joined from outside the local network. 
Please note that we did not checked if the issue also occurs on Fabric version of the Arclight. 

**Error Message:** 
Failed to handle packet ServerboundContainerButtonClickPacket[containerId=3, buttonId=2], suppressing error
java.lang.ClassCastException: class org.bukkit.craftbukkit.v1_21_R1.inventory.CraftInventoryView cannot be cast to class org.bukkit.inventory.view.EnchantmentView (org.bukkit.craftbukkit.v1_21_R1.inventory.CraftInventoryView and org.bukkit.inventory.view.EnchantmentView are in module arclight@1.0.0-SNAPSHOT+d0dda85 of loader 'TRANSFORMER' @2903c6ff)
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.world.inventory.EnchantmentMenu.lambda$slotsChanged$0(EnchantmentMenuMixin_NeoForge.java:600) ~[server-1.21.1-20240808.144430-srg.jar%23178!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.world.inventory.ContainerLevelAccess.lambda$execute$0(ContainerLevelAccessMixin.java:35) ~[server-1.21.1-20240808.144430-srg.jar%23178!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.world.inventory.ContainerLevelAccess$1Anonymous$6bfbecf0d8ac4182833c369bfa0c72aa.evaluate(ContainerLevelAccessMixin.java:39) ~[server-1.21.1-20240808.144430-srg.jar%23178!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.world.inventory.ContainerLevelAccess.execute(ContainerLevelAccessMixin.java:34) ~[server-1.21.1-20240808.144430-srg.jar%23178!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.world.inventory.EnchantmentMenu.slotsChanged(EnchantmentMenuMixin_NeoForge.java:102) ~[server-1.21.1-20240808.144430-srg.jar%23178!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.world.inventory.EnchantmentMenu$1.setChanged(EnchantmentContainer1Mixin.java:40) ~[server-1.21.1-20240808.144430-srg.jar%23178!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.world.SimpleContainer.setItem(SimpleContainerMixin.java:142) ~[server-1.21.1-20240808.144430-srg.jar%23178!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.world.inventory.EnchantmentMenu.lambda$clickMenuButton$1(EnchantmentMenuMixin_NeoForge.java:169) ~[server-1.21.1-20240808.144430-srg.jar%23178!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.world.inventory.ContainerLevelAccess.lambda$execute$0(ContainerLevelAccessMixin.java:35) ~[server-1.21.1-20240808.144430-srg.jar%23178!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.world.inventory.ContainerLevelAccess$1Anonymous$6bfbecf0d8ac4182833c369bfa0c72aa.evaluate(ContainerLevelAccessMixin.java:39) ~[server-1.21.1-20240808.144430-srg.jar%23178!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.world.inventory.ContainerLevelAccess.execute(ContainerLevelAccessMixin.java:34) ~[server-1.21.1-20240808.144430-srg.jar%23178!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.world.inventory.EnchantmentMenu.clickMenuButton(EnchantmentMenuMixin_NeoForge.java:161) ~[server-1.21.1-20240808.144430-srg.jar%23178!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.network.ServerGamePacketListenerImpl.handleContainerButtonClick(ServerPlayNetHandlerMixin_NeoForge.java:1719) ~[server-1.21.1-20240808.144430-srg.jar%23178!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.network.protocol.game.ServerboundContainerButtonClickPacket.handle(ServerboundContainerButtonClickPacket.java:23) ~[server-1.21.1-20240808.144430-srg.jar%23178!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.network.protocol.game.ServerboundContainerButtonClickPacket.handle(ServerboundContainerButtonClickPacket.java:9) ~[server-1.21.1-20240808.144430-srg.jar%23178!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.network.protocol.PacketUtils.md9fa763$lambda$ensureRunningOnSameThread$0$0(PacketThreadUtilMixin.java:538) ~[server-1.21.1-20240808.144430-srg.jar%23178!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.TickTask.run(TickTask.java:18) ~[server-1.21.1-20240808.144430-srg.jar%23178!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoopMixin.java:148) ~[server-1.21.1-20240808.144430-srg.jar%23178!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:23) ~[server-1.21.1-20240808.144430-srg.jar%23178!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.doRunTask(MinecraftServerMixin_NeoForge.java:872) ~[server-1.21.1-20240808.144430-srg.jar%23178!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.doRunTask(MinecraftServerMixin_NeoForge.java:170) ~[server-1.21.1-20240808.144430-srg.jar%23178!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoopMixin.java:122) ~[server-1.21.1-20240808.144430-srg.jar%23178!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServerMixin_NeoForge.java:855) ~[server-1.21.1-20240808.144430-srg.jar%23178!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.pollTask(MinecraftServerMixin_NeoForge.java:849) ~[server-1.21.1-20240808.144430-srg.jar%23178!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoopMixin.java:132) ~[server-1.21.1-20240808.144430-srg.jar%23178!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.managedBlock(MinecraftServerMixin_NeoForge.java:821) ~[server-1.21.1-20240808.144430-srg.jar%23178!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.waitUntilNextTick(MinecraftServerMixin_NeoForge.java:826) ~[server-1.21.1-20240808.144430-srg.jar%23178!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.runServer(MinecraftServerMixin_NeoForge.java:712) ~[server-1.21.1-20240808.144430-srg.jar%23178!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.lambda$spin$2(MinecraftServerMixin_NeoForge.java:267) ~[server-1.21.1-20240808.144430-srg.jar%23178!/:?]
	at java.base/java.lang.Thread.run(Thread.java:1583) [?:?]

### Step to reproduce

1. Create and run the server on Arclight version 1.21.1 (Feudal Kings) with commit d0dda85
2. Enter the server
3. Place Anvil and/or Enchanting Table
4. Attempt to use the blocks.
5. Observe the issue.

### Logs

[latest.log](https://github.com/user-attachments/files/17267644/latest.log)

**Comment by CaptainGolem** [comment]
Bump @IzzelAliz

**Comment by cptDeeJay** [comment]
The Issue is **Verified Fixed** on Commit [35bf457](https://github.com/IzzelAliz/Arclight/commit/35bf457aef2586d1c7a54c592bcfb590243ab6e9)

Thank you very much for your work, and fixing the issue so fast.
